File: Backend/admin/src/main/java/com/shipfast/admin/entity/Branch.java
package com.shipfast.admin.entity;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "branches")
public class Branch {

    @Id
    private String branchId;

    private String name;
    private String type;
    private String address;
    private String state;
    private String managerUserId;
    private String managerName;
    private String contact;
    private Integer staffCount;
    private String status;
    private String description;
}

File: mern-backend/src/models/Branch.js
import { createModel } from '../db/mongo.js';

const BranchMethods = {
  toDto: function () {
    return {
      id: this.branchId,
      branchId: this.branchId,
      name: this.name,
      type: this.type || 'Branch',
      location: this.address || this.location || '',
      address: this.address || this.location || '',
      state: this.state || '',
      manager: this.managerName || this.managerUserId || '',
      managerName: this.managerName || '',
      managerUserId: this.managerUserId || '',
      contact: this.contact || '',
      staffCount: this.staffCount || 0,
      shipmentVolume: this.shipmentVolume || 0,
      revenue: this.revenue || 0,
      performanceScore: this.performanceScore || 100,
      status: this.status || 'Active',
      description: this.description || '',
      createdAt: this.createdAt,
      updatedAt: this.updatedAt
    };
  }
};

const BranchDefaults = {
  type: 'Branch',
  address: '',
  location: '',
  state: '',
  managerName: '',
  managerUserId: '',
  contact: '',
  staffCount: 0,
  shipmentVolume: 0,
  revenue: 0,
  performanceScore: 100,
  status: 'Active',
  description: ''
};

export const Branch = createModel('branches', BranchMethods, BranchDefaults);
